src/Microsoft.TemplateEngine.Cli/Commands/create/TemplateCommand.cs

```diff
@@ -154,27 +154,40 @@ internal class TemplateCommand : Command
         internal async Task<NewCommandStatus> InvokeAsync(ParseResult parseResult, ITelemetryLogger telemetryLogger, CancellationToken cancellationToken)
         {
             TemplateCommandArgs args = new TemplateCommandArgs(this, _instantiateCommand, parseResult);
-
             TemplateInvoker invoker = new TemplateInvoker(_environmentSettings, telemetryLogger, () => Console.ReadLine() ?? string.Empty, _instantiateCommand.Callbacks);
-            if (!args.NoUpdateCheck)
-            {
-                TemplatePackageCoordinator packageCoordinator = new TemplatePackageCoordinator(telemetryLogger, _environmentSettings, _templatePackageManager);
-                Task<CheckUpdateResult?> checkForUpdateTask = packageCoordinator.CheckUpdateForTemplate(args.Template, cancellationToken);
-                Task<NewCommandStatus> instantiateTask = invoker.InvokeTemplateAsync(args, cancellationToken);
-                await Task.WhenAll(checkForUpdateTask, instantiateTask).ConfigureAwait(false);
+            TemplatePackageCoordinator packageCoordinator = new TemplatePackageCoordinator(telemetryLogger, _environmentSettings, _templatePackageManager);
 
-                if (checkForUpdateTask?.Result != null)
-                {
-                    // print if there is update for this template
-                    packageCoordinator.DisplayUpdateCheckResult(checkForUpdateTask.Result, args);
-                }
-                // return creation result
-                return instantiateTask.Result;
+            Task<NewCommandStatus> instantiateTask = invoker.InvokeTemplateAsync(args, cancellationToken);
+            Task<(string Id, string Version)> builtInPackageCheck = packageCoordinator.ValidatePackageAvailabilityAsync(args.Template, cancellationToken);
+            Task<CheckUpdateResult?>? checkForUpdateTask = args.NoUpdateCheck ? null : packageCoordinator.CheckUpdateForTemplate(args.Template, cancellationToken);
```

Comment:
`Task.CompletedTask` feels as better alternative to `null` here (no need for conditional logic below then).

Reply:
Or actually `Task.FromResult<CheckUpdateResult?>(null)` - as it has return.
Not so clean already, but still avoids some extra null checks below

Reply:
I refactored this code and moved check for option to `CheckUpdateForTemplate` method. Now it looks more clean.